packages/server/src/core/initTRPC.ts

```diff
@@ -44,19 +44,32 @@ type CreateRootConfigTypesFromPartial<TTypes extends PartialRootConfigTypes> =
  * - Doesn't need to be a class but it doesn't really hurt either
  */
 
+interface TRPCBuilderConfig<TParams extends PartialRootConfigTypes> {
+  defaultMeta?: TParams['meta'];
+}
+
 class TRPCBuilder<TParams extends PartialRootConfigTypes = object> {
+  constructor(private readonly config: TRPCBuilderConfig<TParams>) {}
+
   context<
     TNewContext extends
       | RootConfigTypes['ctx']
       | ((...args: unknown[]) => RootConfigTypes['ctx']),
   >() {
-    return new TRPCBuilder<
-      FlatOverwrite<TParams, { ctx: Unwrap<TNewContext> }>
-    >();
+    type NextParams = FlatOverwrite<TParams, { ctx: Unwrap<TNewContext> }>;
+
+    return new TRPCBuilder<NextParams>(this.config as NextParams);
   }
-  meta<TNewMeta extends RootConfigTypes['meta']>() {
-    return new TRPCBuilder<FlatOverwrite<TParams, { meta: TNewMeta }>>();
+
+  meta<TNewMeta extends RootConfigTypes['meta']>(defaultMeta?: TNewMeta) {
```

Comment:
A v11 change could be to make defaultMeta mandatory, for now maintain existing behaviour or starting out undefined